resolwe_bio/processes/workflows/bbduk_star_featurecounts_qc.py

```diff
@@ -0,0 +1,802 @@
+            )
+
+        class QualityControl:
+            """Quality control."""
+
+            rrna_reference = DataField(
+                "index:star",
+                label="Indexed rRNA reference sequence",
+                description="Reference sequence index prepared by STAR aligner indexing tool.",
+                required=False,
+            )
+            globin_reference = DataField(
+                "index:star",
+                label="Indexed Globin reference sequence",
+                description="Reference sequence index prepared by STAR aligner indexing tool.",
+                required=False,
+            )
+
+        preprocessing = GroupField(
+            Preprocessing,
+            label="Preprocessing with BBDuk",
+        )
+        alignment = GroupField(
+            Alignment,
+            label="Alignment with STAR",
+        )
+        quantification = GroupField(
+            Quantification,
+            label="Quantification with featureCounts",
+        )
+        downsampling = GroupField(
+            Downsampling,
+            label="Downsampling with Seqtk",
+        )
+        quality_control = GroupField(
+            QualityControl,
+            label="Quality control",
+        )
+
+    class Output:
+        """Output fields."""
+
+    def run(self, inputs, outputs):
+        """Run the workflow."""
+
+        input_bbduk = {
+            "reads": inputs.preprocessing.reads,
+            "min_length": inputs.preprocessing.min_length,
+            "show_advanced": True,
+            "reference": {
+                "sequences": inputs.preprocessing.adapters or [],
+                "literal_sequences": inputs.preprocessing.custom_adapter_sequences,
+            },
+            "processing": {
+                "kmer_length": inputs.preprocessing.kmer_length,
+                "hamming_distance": inputs.preprocessing.hamming_distance,
+            },
+            "operations": {
+                "k_trim": "r"
+                if inputs.preprocessing.adapters
+                or inputs.preprocessing.custom_adapter_sequences
+                else "f",
+                "min_k": inputs.preprocessing.min_k
+                if inputs.preprocessing.adapters
+                or inputs.preprocessing.custom_adapter_sequences
+                else -1,
+                "quality_trim": "r",
+                "trim_quality": inputs.preprocessing.trim_quality,
+                "quality_encoding_offset": inputs.preprocessing.quality_encoding_offset,
+                "ignore_bad_quality": inputs.preprocessing.ignore_bad_quality,
+                "maxns": inputs.preprocessing.maxns,
+            },
+        }
+
+        if inputs.preprocessing.reads.type == "data:reads:fastq:single:":
+            preprocessing = Data.create(
+                process=BioProcess.get_latest(slug="bbduk-single"),
+                input=input_bbduk,
+            )
+        elif inputs.preprocessing.reads.type == "data:reads:fastq:paired:":
```

Comment:
Add also the `else` statement, where the workflow is terminated if the expected input type is not provided.